### PR TITLE
refactor: IWYU fixes and some manual modifications

### PR DESCRIPTION
**Pull request by andiwand**
After using https://github.com/acts-project/acts/pull/2155 I applied the suggested changes and added manual fixes to make the code compile again.

Not sure how customizable IWYU is but it does not work great with Eigen and boost and ipp because it always tries to use the original declaration/definition of the symbol in use. Also it seems not to be able to discover includes necessary for implicit conversions and operator overloads.

### CODE REVIEW

File: Core/include/Acts/Surfaces/ConeBounds.hpp [hunk 1]
```diff
@@ -9,12 +9,19 @@
...
 #include "Acts/Surfaces/SurfaceBounds.hpp"
 #include "Acts/Utilities/detail/periodic.hpp"
 
 #include <array>
+#include <cmath>
+#include <iosfwd>
+#include <stdexcept>
 #include <vector>
 
+#include <stdlib.h>
```
Comment (andiwand) [inline Core/include/Acts/Surfaces/ConeBounds.hpp:23]: todo

File: Core/include/Acts/Propagator/detail/CovarianceEngine.hpp [hunk 2]
```diff
@@ -8,20 +8,22 @@
 
 #pragma once
 
+#include <cmath>
+#include <functional>
+#include <tuple>
+#include <variant>
```
Comment (andiwand) [inline Core/include/Acts/Propagator/detail/CovarianceEngine.hpp:14]: todo

File: Core/include/Acts/Propagator/StraightLineStepper.hpp [hunk 3]
```diff
@@ -8,25 +8,36 @@
 
 #pragma once
 
+#include <algorithm>
+#include <cmath>
+#include <functional>
+#include <limits>
+#include <string>
+#include <tuple>
```
Comment (andiwand) [inline Core/include/Acts/Propagator/StraightLineStepper.hpp:16]: todo

File: Core/include/Acts/Material/VolumeMaterialMapper.hpp [hunk 4]
```diff
@@ -8,6 +8,12 @@
 
 #pragma once
 
+#include <functional>
+#include <map>
+#include <memory>
+#include <utility>
+#include <vector>
```
Comment (andiwand) [inline Core/include/Acts/Material/VolumeMaterialMapper.hpp:15]: todo

File: Core/include/Acts/Material/SurfaceMaterialMapper.hpp [hunk 5]
```diff
@@ -8,6 +8,12 @@
 
 #pragma once
 
+#include <array>
+#include <functional>
+#include <map>
+#include <memory>
+#include <vector>
```
Comment (andiwand) [inline Core/include/Acts/Material/SurfaceMaterialMapper.hpp:15]: todo

**Review by andiwand — COMMENTED** [review]
part 1

**Comment by andiwand** [comment]
I went through all the changes and did a few more manual modifications. Apart from that we can let the CI judge because a full review is very heavy. Ready to be reviewed / merged.

**Review by paulgessinger — APPROVED** [review]
Didn't look at every single file, but from what I've seen this looks good.

**Pull request merged**